generator/lib/control.ml

```diff
@@ -419,56 +451,45 @@ let api =
           methods = [
             {
               Method.name = "probe";
-              description = "[probe uri]: looks for existing SRs on the storage device";
+              description = String.concat " " [
+                "[probe configuration]: can be used iteratively to narrow down configurations";
+                "to use with SR.create, or to find existing SRs on the backing storage";
+              ];
               inputs = [
-                uri;
+                configuration;
               ];
               outputs = [
                { Arg.name = "result";
-                 ty = 
-                   Type.Struct (
-                     ("srs", Type.(Array sr_stat), "SRs found on this storage device"), [
-                      "uris", Type.(Array (Basic String)), "Other possible URIs which may contain SRs"
-                   ]);
+                 ty = Type.(Array (Name "probe_result"));
                  description = "Contents of the storage device";
                }
               ];
             };
             {
               Method.name = "create";
-              description = "[create uuid uri name description configuration]: creates a fresh SR";
+              description = "[create uuid configuration name description]: creates a fresh SR";
```

Comment:
Should we remove the `uuid` param from here too to make it work the same way as `attach`? (So we don't have to duplicate the UUID that's already in the `configuration` returned by probe.)

Reply:
no, we should not. There is no uuid in the configuration for create. Probe can't give us one as there is no SR in the probe result when creating (or only an old one if we're overwriting).